<center>
    <img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/Logos/organization_logo/organization_logo.png" width="300" alt="cognitiveclass.ai logo"  />
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Peer Assignment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a Db2 database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. Click on the links below to download and save the datasets (.CSV files):

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Census Data</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Public Schools</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Crime Data</a>

**NOTE:** Ensure you have downloaded the datasets using the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


### Store the datasets in database tables

To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables in as under:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**

The following required modules are pre-installed in the Skills Network Labs environment. However if you run this notebook commands in a different Jupyter environment (e.g. Watson Studio or Ananconda) you may need to install these libraries by removing the `#` sign before `!pip` in the code cell below.


In [2]:
# These libraries are pre-installed in SN Labs. If running in another environment please uncomment lines below to install them:
!pip install --force-reinstall ibm_db==3.1.0 ibm_db_sa==0.3.3
# Ensure we don't load_ext with sqlalchemy>=1.4 (incompadible)
!pip uninstall sqlalchemy==1.4 -y && pip install sqlalchemy==1.3.24
!pip install ipython-sql

  Using cached ibm_db-3.1.0-py3-none-any.whl
  Using cached ibm_db_sa-0.3.3-py3-none-any.whl
  Using cached SQLAlchemy-1.4.29-cp38-cp38-win_amd64.whl (1.5 MB)
  Using cached greenlet-1.1.2-cp38-cp38-win_amd64.whl (101 kB)
  Attempting uninstall: greenlet
    Found existing installation: greenlet 1.1.2
    Uninstalling greenlet-1.1.2:
      Successfully uninstalled greenlet-1.1.2
  Attempting uninstall: sqlalchemy
    Found existing installation: SQLAlchemy 1.3.24
    Uninstalling SQLAlchemy-1.3.24:
      Successfully uninstalled SQLAlchemy-1.3.24
  Attempting uninstall: ibm-db-sa
    Found existing installation: ibm-db-sa 0.3.3
    Uninstalling ibm-db-sa-0.3.3:
      Successfully uninstalled ibm-db-sa-0.3.3
  Attempting uninstall: ibm-db
    Found existing installation: ibm-db 3.1.0
    Uninstalling ibm-db-3.1.0:
      Successfully uninstalled ibm-db-3.1.0
Found existing installation: SQLAlchemy 1.4.29
Uninstalling SQLAlchemy-1.4.29:
  Successfully uninstalled SQLAlchemy-1.4.29
  Using

In [3]:
%load_ext sql

In [4]:
import csv, sqlite3

con = sqlite3.connect("my_data.db")
cur = con.cursor()

In [6]:
!pip install -q pandas==1.1.5

In [7]:
import pandas
df = pandas.read_csv("https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv")
df.to_sql("CENSUS_DATA", con, if_exists='replace', index=False,method="multi")

df = pandas.read_csv("https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv")
df.to_sql("CHICAGO_CRIME_DATA", con, if_exists='replace', index=False, method="multi")

df = pandas.read_csv("https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv")
df.to_sql("CHICAGO_PUBLIC_SCHOOLS_DATA", con, if_exists='replace', index=False, method="multi")



C:\Users\C0780524\Anaconda3\lib\site-packages\pandas\core\generic.py:2779: UserWarning: The spaces in these column names will not be changed. In pandas versions < 0.14, spaces were converted to underscores.
  name       (index) object 'falcon' 'parrot' 'lion' 'monkey'


In [8]:
%sql sqlite:///my_data.db


## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [12]:
%sql select count(*)  from chicago_crime_data; 

 * sqlite:///my_data.db
Done.


[(533,)]

### Problem 2

##### List community areas with per capita income less than 11000.


In [23]:
%sql select  community_area_number, community_area_name, per_capita_income from CENSUS_DATA  where  per_capita_income <11000

 * sqlite:///my_data.db
Done.


[(26.0, 'West Garfield Park', 10934),
 (30.0, 'South Lawndale', 10402),
 (37.0, 'Fuller Park', 10432),
 (54.0, 'Riverdale', 8201)]

### Problem 3

##### List all case numbers for crimes  involving minors?(children are not considered minors for the purposes of crime analysis)


In [91]:
%%sql 
select   id,case_number, date, description
from chicago_crime_data 
where  DESCRIPTION like '%MINOR%'


 * sqlite:///my_data.db
Done.


[(3987219, 'HL266884', '2005-03-31', 'SELL/GIVE/DEL LIQUOR TO MINOR'),
 (3266814, 'HK238408', '2004-03-13', 'ILLEGAL CONSUMPTION BY MINOR')]

### Problem 4

##### List all kidnapping crimes involving a child?


In [95]:
%%sql 
select id,case_number, date,primary_type , description
from chicago_crime_data 
where primary_type like '%KIDNAPPING%' 
and description like '%CHILD%'; 

 * sqlite:///my_data.db
Done.


[(5276766, 'HN144152', '2007-01-26', 'KIDNAPPING', 'CHILD ABDUCTION/STRANGER')]

### Problem 5

##### What kinds of crimes were recorded at schools?


In [75]:
%%sql 
select distinct primary_type  as CRIME_TYPE 
from chicago_crime_data 
where location_description like '%SCHOOL%'

 * sqlite:///my_data.db
Done.


[('BATTERY',),
 ('CRIMINAL DAMAGE',),
 ('NARCOTICS',),
 ('ASSAULT',),
 ('CRIMINAL TRESPASS',),
 ('PUBLIC PEACE VIOLATION',)]

### Problem 6

##### List the average safety score for all types of schools.


In [98]:
%%sql 
select "Elementary, Middle, or High School" ,  AVG(SAFETY_SCORE)  
from chicago_public_schools_data
group by "Elementary, Middle, or High School" 

 * sqlite:///my_data.db
Done.


[('ES', 49.52038369304557), ('HS', 49.62352941176471), ('MS', 48.0)]

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [76]:
%%sql 
select b.* from 
(select  community_area_number, 
         community_area_name,
         percent_households_below_poverty  
 from census_data order by percent_households_below_poverty desc) b  limit 5

 * sqlite:///my_data.db
Done.


[(54.0, 'Riverdale', 56.5),
 (37.0, 'Fuller Park', 51.2),
 (68.0, 'Englewood', 46.6),
 (29.0, 'North Lawndale', 43.1),
 (27.0, 'East Garfield Park', 42.4)]

### Problem 8

##### Which community area is most crime prone?


In [101]:
%%sql 
select community_area_number, kount most_crime 
from (select community_area_number, count(*) as kount 
      from chicago_crime_data 
      group by community_area_number
      order by kount desc
      limit 1
     )

 * sqlite:///my_data.db
Done.


[(25.0, 43)]

Double-click **here** for a hint

<!--
Query for the 'community area number' that is most crime prone.
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [82]:
%%sql  
select community_area_name 
from  ( select community_area_name , max(hardship_index) from census_data  )

 * sqlite:///my_data.db
Done.


[('Riverdale',)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [133]:
%%sql  
select  cd.community_area_number, community_area_name  , cases
from census_data cd,    (select community_area_number , cases
                               from 
                                  (select community_area_number, count(case_number) cases
                                   from CHICAGO_CRIME_DATA
                                   group by community_area_number
                                   order by count(case_number) desc
                                   limit 1
                                  )       ) cc
where cd.community_area_number=cc.community_area_number
        
            
                                

 * sqlite:///my_data.db
Done.


[(25.0, 'Austin', 43)]

In [122]:
%%sql
select * from CENSUS_DATA--CHICAGO_CRIME_DATA


 * sqlite:///my_data.db
Done.


[(1.0, 'Rogers Park', 7.7, 23.6, 8.7, 18.2, 27.5, 23939, 39.0),
 (2.0, 'West Ridge', 7.8, 17.2, 8.8, 20.8, 38.5, 23040, 46.0),
 (3.0, 'Uptown', 3.8, 24.0, 8.9, 11.8, 22.2, 35787, 20.0),
 (4.0, 'Lincoln Square', 3.4, 10.9, 8.2, 13.4, 25.5, 37524, 17.0),
 (5.0, 'North Center', 0.3, 7.5, 5.2, 4.5, 26.2, 57123, 6.0),
 (6.0, 'Lake View', 1.1, 11.4, 4.7, 2.6, 17.0, 60058, 5.0),
 (7.0, 'Lincoln Park', 0.8, 12.3, 5.1, 3.6, 21.5, 71551, 2.0),
 (8.0, 'Near North Side', 1.9, 12.9, 7.0, 2.5, 22.6, 88669, 1.0),
 (9.0, 'Edison Park', 1.1, 3.3, 6.5, 7.4, 35.3, 40959, 8.0),
 (10.0, 'Norwood Park', 2.0, 5.4, 9.0, 11.5, 39.5, 32875, 21.0),
 (11.0, 'Jefferson Park', 2.7, 8.6, 12.4, 13.4, 35.5, 27751, 25.0),
 (12.0, 'Forest Glen', 1.1, 7.5, 6.8, 4.9, 40.5, 44164, 11.0),
 (13.0, 'North Park', 3.9, 13.2, 9.9, 14.4, 39.0, 26576, 33.0),
 (14.0, 'Albany Park', 11.3, 19.2, 10.0, 32.9, 32.0, 21323, 53.0),
 (15.0, 'Portage Park', 4.1, 11.6, 12.6, 19.3, 34.0, 24336, 35.0),
 (16.0, 'Irving Park', 6.3, 13.1, 10.0, 22.4, 31.6, 27249, 34.0),
 (17.0, 'Dunning', 5.2, 10.6, 10.0, 16.2, 33.6, 26282, 28.0),
 (18.0, 'Montclaire', 8.1, 15.3, 13.8, 23.5, 38.6, 22014, 50.0),
 (19.0, 'Belmont Cragin', 10.8, 18.7, 14.6, 37.3, 37.3, 15461, 70.0),
 (20.0, 'Hermosa', 6.9, 20.5, 13.1, 41.6, 36.4, 15089, 71.0),
 (21.0, 'Avondale', 6.0, 15.3, 9.2, 24.7, 31.0, 20039, 42.0),
 (22.0, 'Logan Square', 3.2, 16.8, 8.2, 14.8, 26.2, 31908, 23.0),
 (23.0, 'Humboldt park', 14.8, 33.9, 17.3, 35.4, 38.0, 13781, 85.0),
 (24.0, 'West Town', 2.3, 14.7, 6.6, 12.9, 21.7, 43198, 10.0),
 (25.0, 'Austin', 6.3, 28.6, 22.6, 24.4, 37.9, 15957, 73.0),
 (26.0, 'West Garfield Park', 9.4, 41.7, 25.8, 24.5, 43.6, 10934, 92.0),
 (27.0, 'East Garfield Park', 8.2, 42.4, 19.6, 21.3, 43.2, 12961, 83.0),
 (28.0, 'Near West Side', 3.8, 20.6, 10.7, 9.6, 22.2, 44689, 15.0),
 (29.0, 'North Lawndale', 7.4, 43.1, 21.2, 27.6, 42.7, 12034, 87.0),
 (30.0, 'South Lawndale', 15.2, 30.7, 15.8, 54.8, 33.8, 10402, 96.0),
 (31.0, 'Lower West Side', 9.6, 25.8, 15.8, 40.7, 32.6, 16444, 76.0),
 (32.0, 'Loop', 1.5, 14.7, 5.7, 3.1, 13.5, 65526, 3.0),
 (33.0, 'Near South Side', 1.3, 13.8, 4.9, 7.4, 21.8, 59077, 7.0),
 (34.0, 'Armour Square', 5.7, 40.1, 16.7, 34.5, 38.3, 16148, 82.0),
 (35.0, 'Douglas', 1.8, 29.6, 18.2, 14.3, 30.7, 23791, 47.0),
 (36.0, 'Oakland', 1.3, 39.7, 28.7, 18.4, 40.4, 19252, 78.0),
 (37.0, 'Fuller Park', 3.2, 51.2, 33.9, 26.6, 44.9, 10432, 97.0),
 (38.0, 'Grand Boulevard', 3.3, 29.3, 24.3, 15.9, 39.5, 23472, 57.0),
 (39.0, 'Kenwood', 2.4, 21.7, 15.7, 11.3, 35.4, 35911, 26.0),
 (40.0, 'Washington Park', 5.6, 42.1, 28.6, 25.4, 42.8, 13785, 88.0),
 (41.0, 'Hyde Park', 1.5, 18.4, 8.4, 4.3, 26.2, 39056, 14.0),
 (42.0, 'Woodlawn', 2.9, 30.7, 23.4, 16.5, 36.1, 18672, 58.0),
 (43.0, 'South Shore', 2.8, 31.1, 20.0, 14.0, 35.7, 19398, 55.0),
 (44.0, 'Chatham', 3.3, 27.8, 24.0, 14.5, 40.3, 18881, 60.0),
 (45.0, 'Avalon Park', 1.4, 17.2, 21.1, 10.6, 39.3, 24454, 41.0),
 (46.0, 'South Chicago', 4.7, 29.8, 19.7, 26.6, 41.1, 16579, 75.0),
 (47.0, 'Burnside', 6.8, 33.0, 18.6, 19.3, 42.7, 12515, 79.0),
 (48.0, 'Calumet Heights', 2.1, 11.5, 20.0, 11.0, 44.0, 28887, 38.0),
 (49.0, 'Roseland', 2.5, 19.8, 20.3, 16.9, 41.2, 17949, 52.0),
 (50.0, 'Pullman', 1.5, 21.6, 22.8, 13.1, 38.6, 20588, 51.0),
 (51.0, 'South Deering', 4.0, 29.2, 16.3, 21.0, 39.5, 14685, 65.0),
 (52.0, 'East Side', 6.8, 19.2, 12.1, 31.9, 42.8, 17104, 64.0),
 (53.0, 'West Pullman', 3.3, 25.9, 19.4, 20.5, 42.1, 16563, 62.0),
 (54.0, 'Riverdale', 5.8, 56.5, 34.6, 27.5, 51.5, 8201, 98.0),
 (55.0, 'Hegewisch', 3.3, 17.1, 9.6, 19.2, 42.9, 22677, 44.0),
 (56.0, 'Garfield Ridge', 2.6, 8.8, 11.3, 19.3, 38.1, 26353, 32.0),
 (57.0, 'Archer Heights', 8.5, 14.1, 16.5, 35.9, 39.2, 16134, 67.0),
 (58.0, 'Brighton Park', 14.4, 23.6, 13.9, 45.1, 39.3, 13089, 84.0),
 (59.0, 'McKinley Park', 7.2, 18.7, 13.4, 32.9, 35.6, 16954, 61.0),
 (60.0, 'Bridgeport', 4.5, 18.9, 13.7, 22.2, 31.3, 22694, 43.0),
 (61.0, 'New City', 11.9, 29.0, 23.0, 41.5, 38.9, 12765, 91.0),
 (62.0, 'West El

In [118]:
%%sql
select community_area_number 
                               from 
                                  (select community_area_number, count(case_number)
                                   from CHICAGO_CRIME_DATA
                                   group by community_area_number
                                   order by count(case_number) desc
                                   limit 1
                                  )

 * sqlite:///my_data.db
Done.


[(25.0,)]

In [117]:
%%sql
select community_area_number, count(case_number)
                                   from CHICAGO_CRIME_DATA
                                   group by community_area_number
                                   order by count(case_number) desc
                                   limit 1

 * sqlite:///my_data.db
Done.


[(25.0, 43)]

Copyright © 2020 This notebook and its source code are released under the terms of the [MIT License](https://bigdatauniversity.com/mit-license?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ).


## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>

## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>
